frontend/src/api/Socket.ts

```diff
@@ -1,54 +1,92 @@
+/* eslint-disable consistent-return */
 import SockJS from 'sockjs-client';
 import Stomp, { Client, Message } from 'stompjs';
 
 let stompClient: Client;
 
-type User = {
+export type User = {
   nickname: string;
 }
 
 // Variable to hold the current connected state.
-let connected = false;
+let connected: boolean = false;
 
 // Create constants for the connection, subscription, and send message URLs.
 export const SOCKET_ENDPOINT:string = '/api/v1/socket/join-room-endpoint';
 const SUBSCRIBE_URL:string = '/api/v1/socket/subscribe-user';
-const SEND_USER_URL:string = '/api/v1/socket/user';
+const ADD_USER_URL:string = '/api/v1/socket/add-user';
+const DELETE_USER_URL:string = '/api/v1/socket/delete-user';
 
-export const sendUser = (nickname:string) => {
-  if (connected) {
-    stompClient.send(SEND_USER_URL, {}, nickname || 'Anonymous');
+export const isValidNickname = (nickname: string) => nickname.length > 0
+  && !nickname.includes(' ') && nickname.length <= 16;
```

Comment:
Should there be a check to make sure multiple people don't share the same username in a game? You used a HashSet to store the users but it doesn't appear to do anything if the nickname already exists.

Reply:
See the Notes section:
> - Things to handle in the future:
>   - Not allow user to join if the nickname is already taken.

Since it would require restructuring, I think I was planning on waiting on this, especially because we don't have separate rooms set up right now.